projects/buck2/build.mk

```diff
@@ -0,0 +1,50 @@
+# --------------------------------------------------------------------------------
+# Project: Buck2 Build Script
+# Description: This script is used for building and managing the Buck2 project.
+# Author: Christian Abrokwa
+# --------------------------------------------------------------------------------
+
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+$(eval $(call project-define,buck2))
+
+PROJECT_NAME := buck2
+PROJECT_VERSION := 2023-09-01
+
+HOST_OUT_DIR := out/host/bin
+BUILD_HOST_DIR :=build/host
+
+BUCK2_HOST_OUT_TARGET := $(HOST_OUT_DIR)/buck2
+BUCK2_HOST_BUILD_TARGET := $(BUILD_HOST_DIR)
+BUCK2_ZIP_TARGET := $(BUCK2_OUT_TARGET).zip
+
+BUCK2_URL := https://github.com/facebook/$(PROJECT_NAME)/archive/refs/tags/$(PROJECT_VERSION).zip
```

Comment:
I think this is a zip archive containing source code. We need the zst file containing the prebuilt binary, like buck2-x86_64-unknown-linux-gnu.zst

Reply:
Noted. I'll work on that.